Python/Python.sublime-syntax

```diff
@@ -1948,895 +1948,1150 @@ contexts:
-    # Single-line string, unicode or not, starting with a SQL keyword
-    - match: '([uU]?)('')(?={{sql_indicator}})'
+      push: inside-single-quoted-f-string
+
+  inside-single-quoted-f-string:
+    - meta_include_prototype: false
+    - meta_content_scope: meta.string.interpolated.python string.quoted.single.python
+    - include: single-quoted-string-end
+    - include: escaped-f-string-braces
+    - include: escaped-unicode-chars
+    - include: escaped-chars
+    - include: f-string-replacements
+
+  single-quoted-u-strings:
+    - match: ([uU]?)(')
       captures:
         1: storage.type.string.python
         2: meta.string.python string.quoted.single.python punctuation.definition.string.begin.python
-      push:
-        - meta_content_scope: meta.string.python
-        - match: "'"
-          scope: string.quoted.single.python punctuation.definition.string.end.python
-          set: after-expression
-        - include: line-continuation-inside-string
-        - match: ''
-          push: scope:source.sql
-          with_prototype:
-            - match: '(?=''|\n)'
-              pop: true
-            - include: escaped-unicode-char
-            - include: escaped-char
-            - include: line-continuation-inside-string
-            - include: constant-placeholder
+      push: inside-single-quoted-u-string
+
+  inside-single-quoted-u-string:
+    - meta_include_prototype: false
+    - meta_content_scope: meta.string.python string.quoted.single.python
+    - include: single-quoted-string-end
+    - include: single-quoted-u-string-syntax
+
+  single-quoted-u-string-syntax:
+    # Single-line string, unicode or not, starting with a SQL keyword
+    - match: (?={{sql_indicator}})
+      set: inside-single-quoted-sql-u-string
     # Single-line string, unicode or not
-    - match: '([uU]?)('')'
+    - match: (?=\S)
+      set: inside-single-quoted-plain-u-string
+
+  inside-single-quoted-plain-u-string:
+    - meta_include_prototype: false
+    - meta_content_scope: meta.string.python string.quoted.single.python
+    - include: single-quoted-string-end
+    - include: single-quoted-u-string-content
+
+  inside-single-quoted-sql-u-string:
+    - meta_include_prototype: false
+    - meta_content_scope: meta.string.python
+    - match: "'"
+      scope: string.quoted.single.python punctuation.definition.string.end.python
+      set: after-expression
+    - include: line-continuation-inside-string
+    - match: ''
+      push: scope:source.sql
+      with_prototype:
+        - include: single-quoted-string-pop
+        - include: single-quoted-u-string-content
+
+  single-quoted-u-string-content:
+    - include: escaped-unicode-chars
+    - include: escaped-chars
+    - include: string-placeholders
+    - include: string-replacements
+
+###[ STRING CONTENTS ]########################################################
+
+  line-continuation-inside-block-string:
+    - match: \\$
+      scope: punctuation.separator.continuation.line.python
```

Comment:
```suggestion
  line-continuations-inside-block-string:
    - match: \\$
      scope: punctuation.separator.continuation.line.python
```

Shouldn't this be plural since it doesn't pop? The following non-block variant is slightly more difficult because it *can* pop in some situations. Perhaps something related to `line-continuation-or-pop`?

Reply:
Current context names are just copied from previous code, but your suggestion is very correct with the popping vs. non-popping nameing convention.

Reply:
How about naming them `string-continuations` vs. `string-continuation-or-illegal-end`?

Reply:
I like `string-continuations`, but shouldn't it be `string-continuations-or-illegal-end` with an s for the otherone?

Reply:
I guess we could debate on where in a context name plural is to be used and where to use singular endlessly. Already renamed such contexts often, internally because of not being satisfied with my own choices every here and then.

Always tried to keep plural at the end of a name and just use singular if any pattern is popping. Don't say that's a perfect choice so. I'd be fine with your suggestion as well. If we don't care about the capture groups, we could probably also split them to `string-continuations` and `illegal-string-end` completely.

Reply:
TBH, like the idea of splitting it up, best for re-usability reasons of patterns. Pushed a commit with according changes.